### ISSUE TITLE
noSmooth() has to be called again inside draw()

### ISSUE THREAD

**Issue by REAS** [processing/processing#3113]
In the default renderer in 0233, Processing is no longer keeping an image's pixels as sharp edges when noSmooth() is applied. The enlarged image is interpolating between colors. Here's an example.

```
PImage img;
PImage crop;

void setup() {
  size(640, 640);
  noSmooth();
  img = loadImage("sea.jpg");
  frameRate(2);
}

void draw() {
  int min = 2;
  int max = 40;
  int x = int(random(img.width-max));
  int y = int(random(img.height-max));
  int w = int(random(min, max));
  crop = img.get(x, y, w, w);
  image(crop, 0, 0, width, height);
}
```

**Comment by benfry** [comment]
Didn't we give this up earlier—sometime in the 2.x series if I'm not mistaken?

**Comment by REAS** [comment]
The code above works fine in 2.0.3, 2.2.1, and 3.0a5. The current code in GitHub is the only time I have seen it behave differently.

**Comment by benfry** [comment]
Sorry, I must be thinking of resizing the image itself. 

Try putting `noSmooth()` inside `draw()`. It'll probably come back.

**Comment by REAS** [comment]
That did the trick. Shall we close this?

**Comment by benfry** [comment]
Nope, it's a bug. It's just that the drawing surface is getting nuked in between trips through the drawing blocks.

**Comment by benfry** [comment]
Fixed for 3.0a6. Was actually something totally different from what I thought; in fact I'm not sure why it works in 3.0a5, but we're set now.
